fastwhisper_server/fastwhisper_server.py

```diff
@@ -0,0 +1,233 @@
+                beam_size=5
+            )
+        except Exception as e:
+            logger.error(f"Transcription failed: {str(e)}")
+            raise HTTPException(
+                status_code=500,
+                detail=f"Transcription failed: {str(e)}"
+            )
+        
+        # Convert segments to list of dictionaries
+        try:
+            segments_list = []
+            for i, segment in enumerate(segments):
+                segments_list.append(
+                    TranscriptionSegment(
+                        id=i,
+                        start=segment.start,
+                        end=segment.end,
+                        text=segment.text
+                    ).to_dict()
+                )
+        except Exception as e:
+            logger.error(f"Failed to process transcription segments: {str(e)}")
+            raise HTTPException(
+                status_code=500,
+                detail="Server error: Could not process transcription results"
+            )
+        
+        # Prepare response
+        response = {
+            "text": " ".join([seg["text"] for seg in segments_list]),
+            "language": info.language,
+            "segments": segments_list,
+            "model_used": request.model
+        }
+        return response
+        
+    except HTTPException:
+        # Re-raise HTTP exceptions as they're already properly formatted
+        raise
+    except ValidationError as e:
+        # Handle Pydantic validation errors
+        logger.error(f"Validation error: {str(e)}")
+        raise HTTPException(
+            status_code=400,
+            detail=f"Invalid request data: {str(e)}"
+        )
+    except Exception as e:
+        # Log unexpected errors with full details
+        logger.exception("Unexpected error during transcription")
+        raise HTTPException(
+            status_code=500,
+            detail="An unexpected error occurred. Please try again later."
+        )
+        
+    finally:
+        # Always clean up the temporary file
+        if temp_file_path and os.path.exists(temp_file_path):
+            try:
+                os.unlink(temp_file_path)
+            except Exception as e:
+                # Log cleanup errors but don't raise them
+                logger.error(f"Error cleaning up temporary file {temp_file_path}: {str(e)}")
+
+@app.post("/chat/completions")
+async def chat_completions(request: TranscribeRequest):
+    """OpenAI-style compatibility endpoint that proxies to /transcribe."""
+    return await transcribe(request)
+
+# Added for compatibility with openai_dart package which automatically prepends /v1
+@app.post("/v1/chat/completions")
+async def chat_completions_v1(request: TranscribeRequest):
+    """OpenAI-style compatibility endpoint with /v1 prefix that proxies to /transcribe."""
+    return await transcribe(request)
+
+if __name__ == "__main__":
+    uvicorn.run(app, host="0.0.0.0", port=8000) 
```

Comment:
@cyberk1ng let's not open this to the outside world but instead restrict this to 127.0.0.1 for security reasons, and use `8083` to use something less commonly used, see above.